### PR TITLE
Add table oci_file_storage_mount_target. Closes #167

### LINKED ISSUE #167: Add table oci_file_storage_mount_target
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.
ile Storage
Mount Targets

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Test names: [ 'tests/oci_file_storage_mount_target' ]
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_file_storage_mount_target []

PRETEST: tests/oci_file_storage_mount_target

TEST: tests/oci_file_storage_mount_target
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # oci_core_subnet.named_test_resource will be created
  + resource "oci_core_subnet" "named_test_resource" {
      + availability_domain        = (known after apply)
      + cidr_block                 = "10.0.0.0/16"
      + compartment_id             = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
      + defined_tags               = (known after apply)
      + dhcp_options_id            = (known after apply)
      + display_name               = "steampipetest3273"
      + dns_label                  = (known after apply)
      + freeform_tags              = {
          + "Name" = "steampipetest3273"
        }
      + id                         = (known after apply)
      + ipv6cidr_block             = (known after apply)
      + ipv6virtual_router_ip      = (known after apply)
      + prohibit_internet_ingress  = (known after apply)
      + prohibit_public_ip_on_vnic = (known after apply)
      + route_table_id             = (known after apply)
      + security_list_ids          = (known after apply)
      + state                      = (known after apply)
      + subnet_domain_name         = (known after apply)
      + time_created               = (known after apply)
      + vcn_id                     = (known after apply)
      + virtual_router_ip          = (known after apply)
      + virtual_router_mac         = (known after apply)
    }

  # oci_core_vcn.named_test_resource will be created
  + resource "oci_core_vcn" "named_test_resource" {
      + cidr_block               = "10.0.0.0/16"
      + cidr_blocks              = (known after apply)
      + compartment_id           = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
      + default_dhcp_options_id  = (known after apply)
      + default_route_table_id   = (known after apply)
      + default_security_list_id = (known after apply)
      + defined_tags             = (known after apply)
      + display_name             = "steampipetest3273"
      + dns_label                = (known after apply)
      + freeform_tags            = (known after apply)
      + id                       = (known after apply)
      + ipv6cidr_blocks          = (known after apply)
      + is_ipv6enabled           = (known after apply)
      + state                    = (known after apply)
      + time_created             = (known after apply)
      + vcn_domain_name          = (known after apply)
    }

  # oci_file_storage_mount_target.named_test_resource will be created
  + resource "oci_file_storage_mount_target" "named_test_resource" {
      + availability_domain = "TvRS:AP-MUMBAI-1-AD-1"
      + compartment_id      = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
      + defined_tags        = (known after apply)
      + display_name        = "steampipetest3273"
      + export_set_id       = (known after apply)
      + freeform_tags       = {
          + "Name" = "steampipetest3273"
        }
      + hostname_label      = (known after apply)
      + id                  = (known after apply)
      + ip_address          = (known after apply)
      + lifecycle_details   = (known after apply)
      + nsg_ids             = (known after apply)
      + private_ip_ids      = (known after apply)
      + state               = (known after apply)
      + subnet_id           = (known after apply)
      + time_created        = (known after apply)
    }

Plan: 3 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + availability_domain = "TvRS:AP-MUMBAI-1-AD-1"
  + freeform_tags       = {
      + "Name" = "steampipetest3273"
    }
  + region              = "ap-mumbai-1"
  + resource_id         = (known after apply)
  + resource_name       = "steampipetest3273"
  + state               = (known after apply)
  + subnet_id           = (known after apply)
  + tenancy_ocid        = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
oci_core_vcn.named_test_resource: Creating...
oci_core_vcn.named_test_resource: Creation complete after 2s [id=ocid1.vcn.oc1.ap-mumbai-1.amaaaaaa6igdexaa6jw7jomc6hfxliglzs2earnbucfuplc3bep4s7vlcuvq]
oci_core_subnet.named_test_resource: Creating...
oci_core_subnet.named_test_resource: Creation complete after 2s [id=ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaav5urw5j2d5xctowkkyzxo5scnt3imyxntu3bp5amwtmpowbuwvfa]
oci_file_storage_mount_target.named_test_resource: Creating...
oci_file_storage_mount_target.named_test_resource: Still creating... [10s elapsed]
oci_file_storage_mount_target.named_test_resource: Creation complete after 20s [id=ocid1.mounttarget.oc1.ap_mumbai_1.aaaaaa4np2swwhrrmjxw2llqojxwiotboaww25lnmjqwsljrfvqwiljr]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

availability_domain = "TvRS:AP-MUMBAI-1-AD-1"
freeform_tags = tomap({
  "Name" = "steampipetest3273"
})
region = "ap-mumbai-1"
resource_id = "ocid1.mounttarget.oc1.ap_mumbai_1.aaaaaa4np2swwhrrmjxw2llqojxwiotboaww25lnmjqwsljrfvqwiljr"
resource_name = "steampipetest3273"
state = "ACTIVE"
subnet_id = "ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaav5urw5j2d5xctowkkyzxo5scnt3imyxntu3bp5amwtmpowbuwvfa"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "availability_domain": "TvRS:AP-MUMBAI-1-AD-1",
    "display_name": "steampipetest3273",
    "freeform_tags": {
      "Name": "steampipetest3273"
    },
    "id": "ocid1.mounttarget.oc1.ap_mumbai_1.aaaaaa4np2swwhrrmjxw2llqojxwiotboaww25lnmjqwsljrfvqwiljr",
    "lifecycle_state": "ACTIVE",
    "subnet_id": "ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaav5urw5j2d5xctowkkyzxo5scnt3imyxntu3bp5amwtmpowbuwvfa"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "availability_domain": "TvRS:AP-MUMBAI-1-AD-1",
    "display_name": "steampipetest3273",
    "freeform_tags": {
      "Name": "steampipetest3273"
    },
    "id": "ocid1.mounttarget.oc1.ap_mumbai_1.aaaaaa4np2swwhrrmjxw2llqojxwiotboaww25lnmjqwsljrfvqwiljr",
    "lifecycle_state": "ACTIVE",
    "subnet_id": "ocid1.subnet.oc1.ap-mumbai-1.aaaaaaaav5urw5j2d5xctowkkyzxo5scnt3imyxntu3bp5amwtmpowbuwvfa"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "region": "ap-mumbai-1",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest3273"
  }
]
✔ PASSED

POSTTEST: tests/oci_file_storage_mount_target

TEARDOWN: tests/oci_file_storage_mount_target

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  display_name,
  id,
  lifecycle_state as state,
  availability_domain,
  time_created
from
  oci_file_storage_mount_target;

+------------------------------+--------------------------------------------------------------------------------------------+--------+-----------------------+---------------------+
| display_name                 | id                                                                                         | state  | availability_domain   | time_created        |
+------------------------------+--------------------------------------------------------------------------------------------+--------+-----------------------+---------------------+
| MountTarget-20210728-0746-17 | ocid1.mounttarget.oc1.ap_mumbai_1.aaaaaa4np2swwhrymjxw2llqojxwiotboaww25lnmjqwsljrfvqwiljr | ACTIVE | TvRS:AP-MUMBAI-1-AD-1 | 2021-07-28 07:46:21 |
+------------------------------+--------------------------------------------------------------------------------------------+--------+-----------------------+---------------------+
```

## List mount targets that are not active

```sql
select
  display_name,
  id,
  lifecycle_state as state
from
  oci_file_storage_mount_target
where
  lifecycle_state <> 'ACTIVE';

+------------------------------+--------------------------------------------------------------------------------------------+----------+
| display_name                 | id                                                                                         | state    |
+------------------------------+--------------------------------------------------------------------------------------------+----------+
| MountTarget-20210728-0611-08 | ocid1.mounttarget.oc1.ap_mumbai_1.aaaaaa4np2swwfsimjxw2llqojxwiotboaww25lnmjqwsljrfvqwiljr | DELETING |
+------------------------------+--------------------------------------------------------------------------------------------+----------+

```
</details>

### CODE REVIEW

File: docs/tables/oci_file_storage_mount_target.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+select
+  display_name,
+  id,
+  lifecycle_state as state,
+  availability_domain,
+  time_created
+from
+  oci_file_storage_mount_target;
+```
+
```
Comment (bigdatasourav) [inline docs/tables/oci_file_storage_mount_target.md:19]:
```suggestion
```

File: docs/tables/oci_file_storage_mount_target.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## List mount targets that are not active
```
Comment (bigdatasourav) [inline docs/tables/oci_file_storage_mount_target.md:21]:
```suggestion
## List mount targets which are not active
```

File: oci/table_oci_file_storage_mount_target.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+	pagesLeft := true
+	for pagesLeft {
+		response, err := session.FileStorageClient.ListMountTargets(ctx, request)
+		if err != nil {
+			plugin.Logger(ctx).Trace("GetError", err)
+			return nil, err
+		}
+
+		for _, mountTarget := range response.Items {
+			d.StreamLeafListItem(ctx, mountTarget)
```
Comment (bigdatasourav) [inline oci/table_oci_file_storage_mount_target.go:172]:
```suggestion
			d.StreamListItem(ctx, mountTarget)
```

File: oci/table_oci_file_storage_mount_target.go [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+func getFileStorageMountTarget(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getFileStorageMountTarget")
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	zone := plugin.GetMatrixItem(ctx)[matrixKeyZone].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("getFileStorageMountTarget", "Compartment", compartment, "OCI_ZONE", zone)
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+	// Restrict the api call to only root compartment/ per region
```
Comment (bigdatasourav) [inline oci/table_oci_file_storage_mount_target.go:195]:
```suggestion

	// Restrict the api call to only root compartment/ per region
```

File: oci/table_oci_file_storage_mount_target.go [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+
+	// handle empty snapshot id in get call
```
Comment (bigdatasourav) [inline oci/table_oci_file_storage_mount_target.go:200]:
```suggestion
	// handle empty mount target id in get call
```

**Review by bigdatasourav — CHANGES_REQUESTED** [review]
@rajeshbal65 some suggested changes.

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**